main/src/vtr_radar/src/modules/preprocessing/extraction/radar_extraction_module.cpp

```diff
@@ -109,10 +109,12 @@ void RadarExtractionModule::run_(QueryCache &qdata0, OutputCache &,
     publisher_initialized_ = true;
   }
 
-  if(!qdata.scan_msg) //no radar data
+  if((qdata.gyro_msg) && (!qdata.scan_msg))  //no online radar data yet check to see if this is a gyro message
```

Comment:
You should be checking the field you actually using, qdata.radar_data not scan_msg.

Reply:
done